crates/net/network/src/config.rs

```diff
@@ -622,7 +623,9 @@ impl<N: NetworkPrimitives> NetworkConfigBuilder<N> {
         hello_message.port = listener_addr.port();
 
         // set the status
-        let status = Status::spec_builder(&chain_spec, &head).build();
+        // eth69 todo: think about if we want to support eth69 status message when building
+        // NetworkConfig
+        let status = StatusMessage::Legacy(Status::spec_builder(&chain_spec, &head).build());
```

Comment:
ideally, we'd use the `EthVersion` we're running to determine whether `status` should be `Legacy` or `Status69` variant, but it looks like we don't define the version til later https://github.com/paradigmxyz/reth/blob/5c64d5915332593d45ad96f99a7bc395102e2e4a/crates/net/network/src/session/mod.rs#L1046-L1068

we could perhaps default to `Legacy` variant in the `NetworkConfig`. then, use a builder to update `status` to `StatusMessage:Eth69` if the version is `69`, right before the `match handshake`. open to suggestions on this